edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs

```diff
@@ -75,11 +75,42 @@ impl Client {
 				serde_json::from_slice(&*response)
 				.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
 
-			let certificate =
-				native_tls::Certificate::from_pem(certificate.as_bytes())
-				.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
+			let mut server_root_certificate = vec![];
+
+			let mut current_cert = String::new();
+			let mut lines = certificate.lines();
+	
+			if lines.next() != Some("-----BEGIN CERTIFICATE-----") {
+				return Err(Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody("malformed PEM: does not start with BEGIN CERTIFICATE".into())));
+			}	
+			current_cert.push_str("-----BEGIN CERTIFICATE-----\n");
+
+			for line in lines {
+
+				if line == "-----END CERTIFICATE-----" {
+					current_cert.push_str("\n-----END CERTIFICATE-----");
+					let current_cert = std::mem::take(&mut current_cert);
+					let certificate =
+						native_tls::Certificate::from_pem(current_cert.as_bytes())
+						.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
+
+					server_root_certificate.push(certificate);
+				}
+				else if line == "-----BEGIN CERTIFICATE-----" {
+					if !current_cert.is_empty() {
```

Comment:
Where does current_cert get reset after the 1st cert is processed?

Reply:
Line 92